### ISSUE TITLE
Improve usability of hooks

### ISSUE THREAD

**Issue by simonbengtsson** [simonbengtsson/jsPDF-AutoTable#188]
To make it easier to add custom behavior and content to autotable the hooks will be removed in favour of one eventHandler in version 3. The syntax will look something like this:

    doc.autoTable({
        eventHandler: function(event) {
            if (event.type === 'endedPage') {
                doc.text("Page header", 20, 20);
            }
        }
    });

Available events will most likely be the following:
- `parsedCell`, `parsedRow` and `parsedTable` (roughly equivalent to the created* hooks)
- `addingCell` and `addingRow` (roughly equivalent to the draw* hooks)
- `addedCell` and `addedRow` (new)
- `endedPage` (equivalent to the addPageContent hook)

Possible future additions: `startedPage`, `startedTable`, `endedTable`.

**Comment by gadamshetty** [comment]
any inputs on custom fonts would greatly apprecaited

**Comment by simonbengtsson** [comment]
Fonts? If you think you have found a bug please post a new issue. If you have a question stackoverflow is a better medium.

**Comment by razorrazor** [comment]
Hi there!
I have no idea how to convert this lines to the new addPageContent:

```
options['beforePageContent'] = header;
options['afterPageContent'] = footer;
```

Please help!

**Comment by simonbengtsson** [comment]
Hey! You could do something like this:

```js
options['addPageContent'] = function() {
    header();
    footer();
}
```

**Comment by razorrazor** [comment]
No man!
If I use your solution, code breaks at line `pdf.text('Report 1', data.settings.margin.left, 50);`.
On the other hand, if I use:
```
options['addPageContent'] = function() {
    header;  // Note missing parenthesis
    footer;  // Note missing parenthesis
}
```
the resulting pdf is blank. I'm using the new 2.1.0. Any other idea?

**Comment by simonbengtsson** [comment]
You are right! I was a bit hasty, try this:

```js
options['addPageContent'] = function(data) {
    header(data);
    footer(data);
}
```

**Comment by razorrazor** [comment]
You did the trick! Thank you!

**Comment by simonbengtsson** [comment]
Done in master. Will be available in version 3.0.
